### PR TITLE
Add `core::option::unwrap` builtin

### PR DESCRIPTION
**Pull request by zhassan-aws**
I used the following function to test:
```rust
fn foo(o: Option<i32>) -> i32 {
    o.unwrap()
}
```

### CODE REVIEW

**Comment by zhassan-aws** [comment]
I'm not sure if the modifications to the tests are OK given that some of the them have a `-- THIS FILE WAS AUTOMATICALLY GENERATED BY AENEAS` header.

**Comment by sonmarcho** [comment]
Regarding your question: the files with the header `-- THIS FILE WAS AUTOMATICALLY GENERATED BY AENEAS` should be regenerated when running `make test`. The other files are to be manually edited.

**Comment by zhassan-aws** [comment]
> Regarding your question: the files with the header `-- THIS FILE WAS AUTOMATICALLY GENERATED BY AENEAS` should be regenerated when running `make test`. The other files are to be manually edited.

Does this mean that my modifications to https://github.com/AeneasVerif/aeneas/pull/125/files#diff-b29d8d6a5dc8845ac62b3ee04427653b06fffe3bc952808636b5433d64557185 for example are NOT OK (since it has the header you mentioned)?

**Comment by sonmarcho** [comment]
I thought you had run `make test` and it had regenerated this file (because this is the output I expect).
(side note: running `OPTIONS="" make test` makes the regeneration a lot faster, because it deactivates some invariant checks)

**Comment by sonmarcho** [comment]
I will regenerate the tests on my side.
Also, it seems the test I had set up to check that there is no diff in the generated files doesn't work: the CI should have failed on this PR (I opened an issue).

**Comment by zhassan-aws** [comment]
> I thought you had run make test and it had regenerated this file (because this is the output I expect).

I did run `make test`, but for some reason, it didn't regenerate the tests. I tried it now, and it regenerated them. I might have messed something up the previous time.

Should I revert the second commit in this PR in which I manually updated the tests?

**Comment by zhassan-aws** [comment]
It seems that some of the manual edits are necessary. If I revert the second commit, regenerate the tests through `make test`, the Lean build fails:
```
[664/676] Building External.FunsExternal
[665/676] Building BetreeMain.FunsExternal
error: > LEAN_PATH=./.lake/packages/std/.lake/build/lib:./.lake/packages/Qq/.lake/build/lib:./.lake/packages/aesop/.lake/build/lib:./.lake/packages/proofwidgets/.lake/build/lib:./.lake/packages/Cli/.lake/build/lib:./.lake/packages/importGraph/.lake/build/lib:./.lake/packages/mathlib/.lake/build/lib:././../../backends/lean/.lake/build/lib:./.lake/build/lib LD_LIBRARY_PATH=./.lake/build/lib /home/ubuntu/.elan/toolchains/leanprover--lean4---v4.7.0/bin/lean ./././External/FunsExternal.lean -R ././. -o ./.lake/build/lib/External/FunsExternal.olean -i ./.lake/build/lib/External/FunsExternal.ilean -c ./.lake/build/ir/External/FunsExternal.c
error: stdout:
./././External/FunsExternal.lean:20:4: error: 'core.option.Option.unwrap' has already been declared
error: external command `/home/ubuntu/.elan/toolchains/leanprover--lean4---v4.7.0/bin/lean` exited with code 1
error: > LEAN_PATH=./.lake/packages/std/.lake/build/lib:./.lake/packages/Qq/.lake/build/lib:./.lake/packages/aesop/.lake/build/lib:./.lake/packages/proofwidgets/.lake/build/lib:./.lake/packages/Cli/.lake/build/lib:./.lake/packages/importGraph/.lake/build/lib:./.lake/packages/mathlib/.lake/build/lib:././../../backends/lean/.lake/build/lib:./.lake/build/lib LD_LIBRARY_PATH=./.lake/build/lib /home/ubuntu/.elan/toolchains/leanprover--lean4---v4.7.0/bin/lean ./././BetreeMain/FunsExternal.lean -R ././. -o ./.lake/build/lib/BetreeMain/FunsExternal.olean -i ./.lake/build/lib/BetreeMain/FunsExternal.ilean -c ./.lake/build/ir/BetreeMain/FunsExternal.c
error: stdout:
./././BetreeMain/FunsExternal.lean:33:4: error: 'core.option.Option.unwrap' has already been declared
error: external command `/home/ubuntu/.elan/toolchains/leanprover--lean4---v4.7.0/bin/lean` exited with code 1
make: *** [Makefile:31: verify] Error 1
```

**Comment by zhassan-aws** [comment]
@sonmarcho is there anything I should do for this PR?

File: compiler/ExtractBuiltin.ml [hunk 1]
```diff
@@ -554,14 +554,17 @@ let builtin_fun_effects =
       (fun n -> (n, { can_fail = false; stateful = false }))
       no_fail_no_state_funs
   in
+  (* TODO: all the functions registered in the [builtin_funs] above should
+     be considered a not using a state. There is a lot of redundancy
```
Comment (zhassan-aws) [inline compiler/ExtractBuiltin.ml:558]:
```suggestion
     be considered as not using a state. There is a lot of redundancy
```

**Review by sonmarcho — APPROVED** [review]

**Pull request merged**